prowler/__main__.py

```diff
@@ -316,15 +320,19 @@ def prowler():
                 )
 
             if "html" in mode:
-                # TODO: generate HTML here
-                add_html_footer(
-                    global_provider.output_options.output_filename,
-                    global_provider.output_options.output_directory,
+                # Generate HTML Finding Object
+                filename = (
+                    f"{global_provider.output_options.output_directory}/"
+                    f"{global_provider.output_options.output_filename}{html_file_suffix}"
                 )
-                fill_html_overview_statistics(
-                    stats,
-                    global_provider.output_options.output_filename,
-                    global_provider.output_options.output_directory,
+                html_finding = HTML(
+                    findings=finding_outputs,
+                    create_file_descriptor=True,
+                    file_path=filename,
+                )
+                # Write HTML Finding Object to file
+                html_finding.batch_write_data_to_file(
+                    provider=global_provider, stats=stats
```

Comment:
Do you need the whole provider object here?

Reply:
Yes, I finally need the provider object since the header need to access to different attributes for each kind of provider when getting the assessments.